lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,1136 @@
+$OperationalStatus = Get-AnsibleParam -obj $params -name "operational_status" -type "str" -default "offline" -ValidateSet "offline","online"
+$ReadOnly = Get-AnsibleParam -obj $params -name "read_only" -type "bool" -default $true
+$Number = Get-AnsibleParam -obj $params -name "number" -type "int"
+# Set attributes partition
+$SetPartitionStyle = Get-AnsibleParam -obj $params -name "partition_style_set" -type "str" -default "gpt" -ValidateSet "gpt","mbr"
+$DriveLetter = Get-AnsibleParam -obj $params -name "drive_letter" -type "str" -ValidateSet "a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"
+$PartitionAccessPath = Get-AnsibleParam -obj $params -name "access_path" -type "str"
+# Set attributes file system
+$FileSystem = Get-AnsibleParam -obj $params -name "file_system" -type "str" -default "ntfs" -ValidateSet "ntfs","refs"
+$Label = Get-AnsibleParam -obj $params -name "label" -type "str" -default "ansible_disk"
+$AllocUnitSize = Get-AnsibleParam -obj $params -name "allocation_unit_size" -type "int" -default 4 -ValidateSet 4,8,16,32,64
+# Set attributes switches
+$LargeFRS = Get-AnsibleParam -obj $params -name "large_frs" -type "bool" -default $false
+$ShortNames = Get-AnsibleParam -obj $params -name "short_names" -type "bool" -default $false
+$IntegrityStreams = Get-AnsibleParam -obj $params -name "integrity_streams" -type "bool" -default $false
+
+# Create a new result object
+$result = @{
+    changed = $false
+}
+
+# Validate option values
+if ($Size -ne $null) {
+    if ($Size -lt 1) {
+        Fail-Json -obj $result -message "Size option value must be at least 1gb"
+    }
+}
+if ($Number -ne $null) {
+    if ([int32]2147483647 -lt $Number) {
+        Fail-Json -obj $result -message "Number option must be of type int32"
+    }
+}
+if ($PartitionAccessPath -ne $null) {
+    if ((-not ((Get-Partition).AccessPaths -like "$PartitionAccessPath*")) -and (Test-Path $PartitionAccessPath -PathType Container) -and ((Get-Item $PartitionAccessPath).LinkType -eq $null)) {
+        if (-not (Get-ChildItem $PartitionAccessPath)) {
+            $true
+        } else {
+            Fail-Json -obj $result -message "$PartitionAccessPath is not an empty folder (contains files and/or folders)"                                    
+        }
+    } elseif ((Get-Partition).AccessPaths -like "$PartitionAccessPath*") {
+        Fail-Json -obj $result -message "$PartitionAccessPath is already in use as access path by another disk"
+    } elseif (-not (Test-Path $PartitionAccessPath -PathType Container)) {
+        Fail-Json -obj $result -message "$PartitionAccessPath is not a valid path/directory/folder on the target for option SetPartitionStyle"
+    } elseif((Get-Item $PartitionAccessPath).LinkType -ne $null) {
+        Fail-Json -obj $result -message "$PartitionAccessPath is already in use as a link of type $((Get-Item $PartitionAccessPath).LinkType)"
+    }
+}
+
+# Functions
+function Search-Disk {
```

Comment:
This function doesn't really work the way I expect it to - it should cope with missing values far better. Really, if I just choose to pass `number`, that should be enough. Basically only compare if not null. I'm not sure how best to achieve this in powershell, I'm afraid.

Reply:
Hi @willthames , it is enough if you pass a number or a size to the module. You can even pass no size or number to the module. The module will then try to find the disk with the information which is given.
The least information which can be given to select the disk is defined by the select options which have default values:
partition_style_select: raw
operational_status: offline
read_only: true

You only need to change them if you want to find a disk with different properties.
If you pass no size and number option it is possible that more than one disk will be found on the target.